### PR TITLE
feat(TD-7): Functionality and styling to add and render tasks

### PR DESCRIPTION
**Pull request by taiman-siddiqui-sazim**
**Description of change**

- Add form adds a task to the list
- List of tasks gets rendered and persists in local storage

**Associated Trello link** 

https://trello.com/c/GfzNQ7RF

**Screenshot of change**

Screenshot 2024-12-05 at 15-20-55

**Dev notes**

edit, delete, task completion and dynamic task list features have not been implemented yet.

### CODE REVIEW

**Comment by mahir-mohsin-sazim** [comment]
Fix the merge conflicts

File: modules/to-do/containers/ToDoPageContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  useEffect(() => {
+    const storedTasks = localStorage.getItem("tasks");
+    if (storedTasks) {
+      setTasks(JSON.parse(storedTasks));
+    }
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem("tasks", JSON.stringify(tasks));
```
Comment (ishmamur-rahman-sazim) [inline modules/to-do/containers/ToDoPageContainer.tsx:16]: Can we do this below `setTasks` in the `addTask` method? Might reduce the need for an extra useEffect

File: modules/to-do/containers/ToDoPageContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from "react";
+import { AddTask } from "../components/AddTask/AddTask";
+import { TaskList } from "../components/TaskList/TaskList";
+
+export const ToDoPageContainer = () => {
+  const [tasks, setTasks] = useState<{ id: number; title: string; completed: boolean }[]>([]);
```
Comment (ishmamur-rahman-sazim) [inline modules/to-do/containers/ToDoPageContainer.tsx:6]: Looks like we have a definition for `ITask`. Let's import it from there

File: shared/typedefs/interfaces.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export interface Task {
```
Comment (ishmamur-rahman-sazim) [inline shared/typedefs/interfaces.ts:1]: Interface names should be prefixed by an `I`. So in this case, we can name this `ITask`

**Review by ishmamur-rahman-sazim — CHANGES_REQUESTED** [review]

File: modules/to-do/components/AddTask/AddTask.types.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export interface IAddTaskProps {
```
[the remark this answers is not in the document]
Comment (taiman-siddiqui-sazim) [reply modules/to-do/components/AddTask/AddTask.types.ts]: Addressed

**Review by taiman-siddiqui-sazim — COMMENTED** [review]
Addressed.

**Review by mahir-mohsin-sazim — APPROVED** [review]

**Review by ishmamur-rahman-sazim — APPROVED** [review]

**Pull request merged**